code/game/objects/structures/beds_chairs/bed.dm

```diff
@@ -234,3 +234,17 @@
 	name = "double bed"
 	desc = "A luxurious double bed, for those too important for small dreams."
 	icon_state = "bed_double"
+	buildstackamount = 4
+	max_buckled_mobs = 2
+	///The mob who buckled to this bed second, to avoid other mobs getting pixel-shifted before he unbuckles.
+	var/mob/living/goldilocks
+
+/obj/structure/bed/double/post_buckle_mob(mob/living/target)
+	if(buckled_mobs.len > 1 && !goldilocks) //Push the second buckled mob a bit higher from the normal lying position
```

Comment:
```suggestion
	if(length(buckled_mobs) && !goldilocks) //Push the second buckled mob a bit higher from the normal lying position
```
`length` is faster, it also won't runtime in the event of null lists which is cool too.